cannon/mipsevm/multithreaded/mips.go

```diff
@@ -212,7 +212,7 @@ func (m *InstrumentedState) handleSyscall() error {
 	case arch.SysLseek:
 	default:
 		// These syscalls have the same values on 64-bit. So we use if-stmts here to avoid "duplicate case" compiler error for the cannon64 build
-		if arch.IsMips32 && syscallNum == arch.SysFstat64 || syscallNum == arch.SysStat64 || syscallNum == arch.SysLlseek {
+		if arch.IsMips32 && (syscallNum == arch.SysFstat64 || syscallNum == arch.SysStat64 || syscallNum == arch.SysLlseek) {
```

Comment:
I think it's worth adding a test for this.  We would just need to append `arch.UndefinedSysNr` to the `unsupportedSyscalls` [here](https://github.com/ethereum-optimism/optimism/blob/89f623e3d9b9c9591d21750707de89d97288140b/cannon/mipsevm/tests/evm_multithreaded64_test.go#L474).

Reply:
The undefined syscall number is a fully set 64-bit value. This was chosen to avoid the possibility of conflicting syscall numbers as syscalls are 32-bit values.

As such, adding undefined syscalls to the unsupported syscall table requires retyping the syscall numbers in tests. I think this creates confusion in how such syscalls are tested for little benefit. And this is only sorta useful to test the unreachable codepath on cannon64. Because on cannon32, the parenthesis fix doesn't change the actual logic for no-op handling - thus you can't write a cannon32 test for this issue.

Reply:
Added a cannon64 test case for the undefined code path.